### PR TITLE
[platform] Add getting platform specfic Multipass storage location

### PR DESCRIPTION
**Pull request by townsend2010**

### CODE REVIEW

**Review by Saviq — CHANGES_REQUESTED** [review]
```
[ RUN      ] Daemon.data_path_valid
D:\a\multipass-private\multipass-private\tests\test_daemon.cpp(241): error: Expected equality of these values:
  config->data_directory.toStdString()
    Which is: "C:/ProgramData/Multipass/data"
  data_dir.path().toStdString()
    Which is: "C:/Users/runneradmin/AppData/Local/Temp/multipass_tests-fTqDtj"
D:\a\multipass-private\multipass-private\tests\test_daemon.cpp(242): error: Expected equality of these values:
  config->cache_directory.toStdString()
    Which is: "C:/ProgramData/Multipass/cache"
  cache_dir.path().toStdString()
    Which is: "C:/Users/runneradmin/AppData/Local/Temp/multipass_tests-VEPtLi"
D:\a\multipass-private\multipass-private\tests\test_daemon.cpp(231): error: Actual function call count doesn't match EXPECT_CALL(mpt::MockStandardPaths::mock_instance(), writableLocation(mp::StandardPaths::CacheLocation))...
         Expected: to be called once
           Actual: never called - unsatisfied and active
D:\a\multipass-private\multipass-private\tests\test_daemon.cpp(234): error: Actual function call count doesn't match EXPECT_CALL(mpt::MockStandardPaths::mock_instance(), writableLocation(mp::StandardPaths::AppDataLocation))...
         Expected: to be called once
           Actual: never called - unsatisfied and active
[  FAILED  ] Daemon.data_path_valid (3 ms)
```

**Comment by townsend2010** [comment]
Ugh, of course it's just never straight forward....

**Comment by Saviq** [comment]
`Platform::daemon_config_home` needs adapting I believe?

**Comment by townsend2010** [comment]
> `Platform::daemon_config_home` needs adapting I believe?

Not in this PR.

File: tests/test_daemon.cpp [hunk 1]
```diff
@@ -227,20 +227,16 @@ TEST_F(Daemon, proxy_contains_valid_info)
...
 {
-    QTemporaryDir data_dir, cache_dir;
-
-    EXPECT_CALL(mpt::MockStandardPaths::mock_instance(), writableLocation(mp::StandardPaths::CacheLocation))
-        .WillOnce(Return(cache_dir.path()));
+    QTemporaryDir storage_dir;
 
-    EXPECT_CALL(mpt::MockStandardPaths::mock_instance(), writableLocation(mp::StandardPaths::AppDataLocation))
-        .WillOnce(Return(data_dir.path()));
+    EXPECT_CALL(*mock_platform, multipass_storage_location()).WillOnce(Return(storage_dir.path()));
```
Comment (Saviq) [inline tests/test_daemon.cpp:232]: You still want to test these if storage location comes back empty, that's where this coverage drop comes from, isn't it?

https://app.codecov.io/gh/canonical/multipass/compare/2485/changes#D1L117

**Review by Saviq — CHANGES_REQUESTED** [review]

File: tests/test_daemon.cpp (hunk 1 above)
Comment (townsend2010) [reply tests/test_daemon.cpp:232]: Ok, yeah, I need a new test case for this condition.

Comment (townsend2010) [reply tests/test_daemon.cpp]: Ok, really putting it back to (nearly) what it was is better.

**Review by Saviq — APPROVED** [review]
Good!

bors merge

**Comment by townsend2010** [comment]
bors retry

**Comment by townsend2010** [comment]
Oh, duh, right, it's gonna fail on Windows.  Will merge manually.

**Pull request merged**